fs2/src/main/scala/io/finch/fs2/package.scala

```diff
@@ -0,0 +1,62 @@
+package io.finch
+
+import _root_.fs2.Stream
+import cats.effect.Effect
+import com.twitter.finagle.http.Response
+import com.twitter.io.Buf
+import io.finch.internal.futureToEffect
+import io.finch.streaming.StreamFromReader
+import shapeless.Witness
+
+package object fs2 extends StreamInstances {
+
+  implicit def streamFromReader[F[_] : Effect]: StreamFromReader[Stream, F] = StreamFromReader.instance { reader =>
+    Stream
+      .repeatEval(futureToEffect(reader.read()))
```

Comment:
Thinking out loud. Maybe we shouldn't need to change `ToEffect` and instead, wrap the call to it with `F.delay` here?

What I'm getting at is I'm not sure `ToEffect` has to be lazy. As you mentioned, in the `Mapper`, we sort of already wrap it with `() =>`.

Reply:
Done